src/serena/util/shell.py

```diff
@@ -42,6 +43,38 @@ def execute_shell_command(command: str, cwd: str | None = None, capture_stderr:
     return ShellCommandResult(stdout=stdout, stderr=stderr, return_code=process.returncode, cwd=cwd)
 
 
+def safe_execute_shell_command(command: str, cwd: str | None = None, capture_stderr: bool = False) -> ShellCommandResult:
```

Comment:
@opcode81 could you have a look pls? I don't think we need two functions, maybe this should replace the already existing one?